android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt

```diff
@@ -1183,6 +1184,59 @@ class GeckoPromptDelegateTest {
         assertEquals(authRequest.level, AC_AUTH_LEVEL.NONE)
     }
 
+    @Test
+    fun `WHEN onSelectIdentityCredentialProvider is called THEN SelectProvider prompt request must be provided with the correct callbacks`() {
+        val mockSession = GeckoEngineSession(runtime)
+        var selectProviderRequest: IdentityCredential.SelectProvider = mock()
+        var onConfirmWasCalled = false
+        var onDismissWasCalled = false
+
+        val promptDelegate = GeckoPromptDelegate(mockSession)
+        mockSession.register(
+            object : EngineSession.Observer {
+                override fun onPromptRequest(promptRequest: PromptRequest) {
+                    selectProviderRequest = promptRequest as IdentityCredential.SelectProvider
+                }
+            },
+        )
+
+        val geckoProvider = GECKO_PROMPT_PROVIDER_SELECTOR(0, "name", "icon")
+        val acProvider = geckoProvider.toProvider()
+        val geckoPrompt = geckoProviderSelectorPrompt(listOf(geckoProvider))
+        var geckoResult = promptDelegate.onSelectIdentityCredentialProvider(mock(), geckoPrompt)
+
+        geckoResult.accept {
+            onConfirmWasCalled = true
+        }
+
+        with(selectProviderRequest) {
+            // Verifying we are parsing the providers correctly.
+            assertEquals(acProvider, this.providers.first())
+
+            onConfirm(acProvider)
+
+            shadowOf(getMainLooper()).idle()
+            assertTrue(onConfirmWasCalled)
+            whenever(geckoPrompt.isComplete).thenReturn(true)
+
+            // Just making sure we are not completing the geckoResult twice.
+            onConfirmWasCalled = false
+            onConfirm(acProvider)
+            shadowOf(getMainLooper()).idle()
+            assertFalse(onConfirmWasCalled)
+        }
+
+        // Verifying we are handling the dismiss correctly.
+        geckoResult = promptDelegate.onSelectIdentityCredentialProvider(mock(), geckoProviderSelectorPrompt(listOf(geckoProvider)))
```

Comment:
@titooan I figured out what was the issue, I was reusing the `geckoResult`, which was already completed, now I'm just calling  `geckoProviderSelectorPrompt` again.

Reply:
Oh great, well done!